web/impact/impact/permissions/v1_api_permissions.py

```diff
@@ -75,7 +75,7 @@ def has_permission(self, request, view):
 
 class OfficeHourPermission(IsAuthenticated):
     def has_permission(self, request, view):
-        self.message = self.get_message(request)
+        message = self.get_message(request)
```

Comment:
removing the `self.` means that `message` is now a local variable in the `has_permission` function, and is not visible to the `OfficeHourPermission` class, which is not what you want.

This is an interesting case, since the message is being generated dynamically. It would be interesting to try renaming `self.get_message` as `self.message` and tagging it with a `property` decorator. That ought to work, but it'd need to be tested. Or we could go back to calling this `self.message`

In any case, if tests pass with this change, then we don't have tests covering this class's `message`